### PR TITLE
Storage tests

### LINKED ISSUE #92: Prepare unit tests for storage
Prepare unit tests for storage

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Unit tests for `storage.go`

Fixes #92

## Type of change

- Unit tests (no changes in the code)

## Testing steps

Done on CI automatically.

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: storage_test.go [hunk 1]
```diff
@@ -15,3 +15,698 @@ limitations under the License.
...
+	mock.ExpectCommit()
+	mock.ExpectClose()
+
+	// prepare connection to mocked database
+	storage := main.NewFromConnection(connection, 1)
+
+	// call the tested method
+	err := storage.DatabaseDropTables()
+	if err != nil {
+		t.Errorf("error was not expected while cleaning up database: %s", err)
```
Comment (epapbak) [inline storage_test.go:599]: dropping database tables

File: storage_test.go [hunk 2]
```diff
@@ -15,3 +15,698 @@ limitations under the License.
...
+	mock.ExpectCommit()
+	mock.ExpectClose()
+
+	// prepare connection to mocked database
+	storage := main.NewFromConnection(connection, 1)
+
+	// call the tested method
+	err := storage.DatabaseDropIndexes()
+	if err != nil {
+		t.Errorf("error was not expected while cleaning up database: %s", err)
```
Comment (epapbak) [inline storage_test.go:627]: dropping database indexes

**Review by epapbak — APPROVED** [review]
With some small things to update in comments

File: storage_test.go [hunk 3]
```diff
@@ -15,3 +15,698 @@ limitations under the License.
...
+	}
+
+	// connection to mocked DB needs to be closed properly
+	checkConnectionClose(t, connection)
+
+	// check if all expectations were met
+	checkAllExpectations(t, mock)
+}
+
+// TestGetLatestKafkaOffset function checks the method
```
[the remark this answers is not in the document]
Comment (tisnik) [reply storage_test.go]: fixed, thank you

File: storage_test.go (hunk 1 above)
Comment (tisnik) [reply storage_test.go]: fixed, thank you

File: storage_test.go (hunk 2 above)
Comment (tisnik) [reply storage_test.go]: fixed, thank you

**Pull request merged**